aten/src/ATen/native/mps/operations/Eye.mm

```diff
@@ -0,0 +1,86 @@
+#include <ATen/ATen.h>
+#include <ATen/Tensor.h>
+#include <ATen/Utils.h>
+#include <ATen/mps/MPSStream.h>
+#include <ATen/native/mps/OperationUtils.h>
+#include <torch/library.h>
+#include "c10/util/Optional.h"
+
```

Comment:
At the top level can you please outline all the steps to add an Op for MPS backend, for instance:
1. Registering the ops in native_functions with `mps` dispatch key
2. Finding the function interface depending on the `structured` vs `non-structured op.`
3. Add boiler-plate error checking for Torch following the convention of CPU and CUDA...
4. ....

Following up on comments at other places where needed below.